socs/agents/hi6200/agent.py

```diff
@@ -0,0 +1,137 @@
+import argparse
+import time
+
+from ocs import ocs_agent, site_config
+from ocs.ocs_twisted import TimeoutLock
+
+from socs.agents.hi6200.drivers import Hi6200Interface
+
+
+class Hi6200Agent:
+    def __init__(self, agent, ip_address, tcp_port):
+        self.agent = agent
+        self.log = agent.log
+        self.lock = TimeoutLock()
+
+        self.job = None
+        self.ip_address = ip_address
+        self.tcp_port = tcp_port
+        self.monitor = False
+
+        self.scale = None
+
+        # Registers Scale Output
+        agg_params = {
+            'frame_length': 10 * 60,
+        }
+        self.agent.register_feed('scale_output',
+                                 record=True,
+                                 agg_params=agg_params,
+                                 buffer_time=0)
+
+    @ocs_agent.param('_')
+    def init(self, session, params=None):
+        """init()
+
+        **Task** - Initialize connection to the Hi 6200 Weight Sensor.
+
+        """
+        with self.lock.acquire_timeout(0) as acquired:
+            if not acquired:
+                return False, "Could not acquire lock"
+
+            self.scale = Hi6200Interface(self.ip_address, self.tcp_port)
+
+            self.log.info("Connected to scale.")
+
+        return True, 'Initialized Scale.'
+
+    @ocs_agent.param('wait', type=float, default=1)
+    def monitor_weight(self, session, params=None):
+        """
+
+        **Process** - Continuously monitor scale gross and net weights.
+
+        Parameters:
+            wait (float, optional): Time to wait between measurements
+                [seconds].
+
+        """
```

Comment:
I'm thinking of just removing the `session.data` calls from this function. I don't think there's any reason to actually query this instead of just looking at Grafana. 

Is there any point to leaving it in if all I want to do is look at the data on Grafana?
If not, and I remove it, is there any need for the documentation in the docstring?

Reply:
If removed, then no, no need to document. The reason to include `session.data` generally is so other programs can access the information. ocs-web for instance, uses this to display info on a custom page for the Agent.

`session.data` is important to document the structure of because there is no strict form (other than the fact that it's a `dict`), so for authors of OCS control programs that use a client for this Agent they can see what information is available without having to run the Agent.

Reply:
No need for people to pull session data from this agent. I'm removing this functionality.